src/AppBundle/Validator/Constraints/UniquePropertyValidator.php

```diff
@@ -0,0 +1,113 @@
+ *
+ * @DI\Validator("app.validator.unique_property")
+ */
+class UniquePropertyValidator extends ConstraintValidator
+{
+    /**
+     * @var ManagerRegistry
+     */
+    private $managerRegistry;
+
+    /**
+     * Constructor.
+     *
+     * @param ManagerRegistry $managerRegistry
+     *
+     * @DI\InjectParams({"managerRegistry" = @DI\Inject("doctrine")})
+     */
+    public function __construct(ManagerRegistry $managerRegistry)
+    {
+        $this->managerRegistry = $managerRegistry;
+    }
+
+    /**
+     * Validates whether the given value is unique when using it as a specific field on a specific model.
+     *
+     * @param string     $value
+     * @param Constraint $constraint
+     *
+     * @throws UnexpectedTypeException       If the constraint is no UniqueProperty constraint.
+     * @throws UnexpectedTypeException       If the ExecutionContext is invalid (to be removed when upgrading to 3.0, just used in order to verify the correct context).
+     * @throws UnexpectedTypeException       If the value is not a scalar value or has a __toString() interceptor.
+     * @throws ConstraintDefinitionException If the manager alias could not be loaded or there's no manager for a specific class
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        /** @var UniqueProperty $constraint */
+        if (!$constraint instanceof UniqueProperty) {
+            throw new UnexpectedTypeException($constraint, UniqueProperty::class);
+        }
+
+        /** @var ExecutionContextInterface $context */
+        $context = $this->context;
+        if (!$context instanceof ExecutionContextInterface) {
+            throw new UnexpectedTypeException($context, ExecutionContextInterface::class);
+        }
+
+        if (!is_scalar($value)
+            && (!is_object($value) && !method_exists($value, '__toString'))
+        ) {
+            throw new UnexpectedTypeException($value, 'scalar');
+        }
+
+        $value = (string) $value;
+
+        if (null !== $managerAlias = $constraint->manager) {
+            $manager = $this->managerRegistry->getManager($managerAlias);
+
+            if (!$manager) {
+                throw new ConstraintDefinitionException(sprintf('No such entity manager with alias "%s"!', $managerAlias));
+            }
+        } else {
+            $entityAlias = $constraint->entity;
+            $manager     = $this->managerRegistry->getManagerForClass($constraint->entity);
+
+            if (!$manager) {
+                throw new ConstraintDefinitionException(sprintf('Cannot find entity manager for model "%s"!', $entityAlias));
+            }
+        }
+
+        /** @var \Doctrine\Common\Persistence\ObjectRepository $repository */
+        $repository = $manager->getRepository($constraint->entity);
+        $field      = $constraint->field;
+        $query      = [$field => $value];
+        $search     = $repository->findOneBy($query);
+
+        if (!empty($search)) {
```

Comment:
in order to avoid confusion when using invalid fields, it should be validated whether the field exists on the entity